src/verify.rs

```diff
@@ -216,8 +217,7 @@ mod attestation {
         // Get the attestation report signature
         let ar_signature = EcdsaSig::try_from(&att_report.signature)
             .context("Failed to get ECDSA Signature from attestation report.")?;
-        let mut report_bytes = Vec::new();
-        att_report.write_bytes(&mut report_bytes)?;
+        let report_bytes = att_report.to_bytes()?.to_vec();
```

Comment:
Don't need the to_vec
```suggestion
        let report_bytes = att_report.to_bytes()?;
```